app/main.py

```diff
@@ -1,3 +1,44 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        return Distance(self.km + self.instancecheck(other))
+
+    def __iadd__(self, other):
+        self.km += self.instancecheck(other)
+        return self
+
+    def __mul__(self, other):
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km